src/net/client.py

```diff
@@ -147,3 +151,17 @@ async def add_transaction(
         self, tx: Transaction, token: Optional[str] = None
     ) -> Dict[str, int]:
         return await self._gateway.add_transaction(tx, token)
+
+    async def compile_and_deploy_contract(
```

Comment:
I suggest we name it deploy_contract and accept both input_files and compiled_contract. In case compiled_contract is not provided we can compile input_files.

Reply:
Also we could accept contract as a string and compile it as contract.cairo. I imagine it would we useful for simple examples.